pkg/apis/dashboard/migration/conversion/conversion.go

```diff
@@ -0,0 +1,199 @@
+package conversion
+
+import (
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	dashboard "github.com/grafana/grafana/pkg/apis/dashboard"
+	dashboardV0 "github.com/grafana/grafana/pkg/apis/dashboard/v0alpha1"
+	dashboardV1 "github.com/grafana/grafana/pkg/apis/dashboard/v1alpha1"
+	dashboardV2 "github.com/grafana/grafana/pkg/apis/dashboard/v2alpha1"
+)
+
+func ToInternalDashboard(scheme *runtime.Scheme, obj runtime.Object) (*dashboard.Dashboard, error) {
+	dash := &dashboard.Dashboard{}
+	if err := scheme.Convert(obj, dash, nil); err != nil {
+		return nil, err
+	}
+	return dash, nil
+}
+
+func FromInternalDashboard(scheme *runtime.Scheme, dash *dashboard.Dashboard, obj runtime.Object) error {
+	return scheme.Convert(dash, obj, nil)
+}
+
+func RegisterConversions(s *runtime.Scheme) error {
+	if err := s.AddConversionFunc((*dashboard.Dashboard)(nil), (*dashboardV0.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_to_V0(a.(*dashboard.Dashboard), b.(*dashboardV0.Dashboard), scope)
+	}); err != nil {
+		return err
+	}
+	if err := s.AddConversionFunc((*dashboard.Dashboard)(nil), (*dashboardV1.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_to_V1(a.(*dashboard.Dashboard), b.(*dashboardV1.Dashboard), scope)
+	}); err != nil {
+		return err
+	}
+	if err := s.AddConversionFunc((*dashboard.Dashboard)(nil), (*dashboardV2.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_to_V2(a.(*dashboard.Dashboard), b.(*dashboardV2.Dashboard), scope)
+	}); err != nil {
+		return err
+	}
+
+	if err := s.AddConversionFunc((*dashboardV0.Dashboard)(nil), (*dashboard.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_V0_to_Internal(a.(*dashboardV0.Dashboard), b.(*dashboard.Dashboard), scope)
+	}); err != nil {
+		return err
+	}
+	if err := s.AddConversionFunc((*dashboardV0.Dashboard)(nil), (*dashboardV1.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_V0_to_V1(a.(*dashboardV0.Dashboard), b.(*dashboardV1.Dashboard), scope)
+	}); err != nil {
+		return err
+	}
+	if err := s.AddConversionFunc((*dashboardV0.Dashboard)(nil), (*dashboardV2.Dashboard)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_V0_to_V2(a.(*dashboardV0.Dashboard), b.(*dashboardV2.Dashboard), scope)
```

Comment:
are these required due to the changes to versioned storage in `register.go`, or are they an independent optimization? if they are required due to the changes to versioned storage, i think we might be able to simplify and still use the internal version as the hub by wrapping the RESTOptionsGetter and setting a new EncodeVersioner for each version similar to [this](https://github.com/grafana/grafana/blob/2a801af804ccd66d4cb8a0d6b64f356d1cf0f253/pkg/registry/apis/dashboard/register.go#L353-L371)

Reply:
Trying that here -- https://github.com/grafana/grafana/pull/101496 

It still has the weird list vs get behavior 🤷🏻‍♂️

Reply:
right now both version to/from internal and direct version to/from version conversions exist. that seems more complex to me unless we can remove the typed internal version